File: apprefurbished/src/main/java/com/androidcollider/easyfin/common/ui/fragments/common/CommonFragmentWithEvents.java
package com.androidcollider.easyfin.common.ui.fragments.common;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.Nullable;

import org.greenrobot.eventbus.EventBus;

/**
 * @author Ihor Bilous
 */

public abstract class CommonFragmentWithEvents extends AbstractBaseFragment {

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        EventBus.getDefault().register(this);
        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        EventBus.getDefault().unregister(this);
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/common/ui/fragments/common/CommonFragmentWithEvents.kt
package com.androidcollider.easyfin.common.ui.fragments.common

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import org.greenrobot.eventbus.EventBus

/**
 * @author Ihor Bilous
 */
abstract class CommonFragmentWithEvents : AbstractBaseFragment() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        EventBus.getDefault().register(this)
        return super.onCreateView(inflater, container, savedInstanceState)
    }

    override fun onDestroyView() {
        super.onDestroyView()
        EventBus.getDefault().unregister(this)
    }
}